components/VM_Arm/src/main.c

```diff
@@ -1033,13 +1033,15 @@ static vm_frame_t on_demand_iterator(uintptr_t addr, void *cookie)
     /* Attempt allocating device memory */
     err = alloc_vm_device_cap(paddr, vm, &frame_result);
     if (!err) {
+        ZF_LOGE("Created device-backed memory for addr 0x%"PRIxPTR, addr);
```

Comment:
Using `ZF_LOGE` is semantically wrong here, this must be `ZF_LOGI`

Reply:
I'm okay with making this a `ZF_LOGI`. It depends on whether the on-demand info should always be printed or not. Is that information the user always needs to see?

Reply:
The behavior for how and when devices are installed into a vm by the vmm is quite confusing and a main contributor for weird runtime behavior and should be easy to diagnose.

Reply:
I'm going to agree with Kent here. `printf` is used by the VMM in other places to print out important information, and device passthrough is very important. And with the On Demand Device Install, if the device is being backed by RAM, then it may not be working properly, and the user should have at least a hint of what may be causing the problem.

Reply:
Ok, then it's a follow-up task to review the print functions.